catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/HistoryMetacardImpl.java

```diff
@@ -0,0 +1,249 @@
+     * {@link HistoryMetacardImpl} as the sourceMetacard.
+     *
+     * @param sourceMetacard Metacard to clone and create a history item from
+     * @param action         Which action was done to modify the metacard
+     * @throws IllegalArgumentException
+     */
+    public HistoryMetacardImpl(Metacard sourceMetacard, Action action, Subject subject) {
+        super(sourceMetacard, versionHistoryMetacard);
+        if (sourceMetacard instanceof HistoryMetacardImpl) {
+            throw new IllegalArgumentException(
+                    "Cannot create a history item from a history metacard.");
+        }
+
+        this.setState(action);
+        this.setIdHistory(sourceMetacard.getId());
+        this.setTagsHistory(sourceMetacard.getTags());
+
+        String editedBy = SubjectUtils.getEmailAddress(subject);
+        if (editedBy == null || editedBy.equals("")) {
+            editedBy = SubjectUtils.getName(subject);
+        }
+        this.setEditedBy(editedBy);
+
+        this.setVersioned(Date.from(Instant.now()));
+        this.setId(UUID.randomUUID()
+                .toString()
+                .replace("-", ""));
+        this.setTags(Collections.singleton(HISTORY_TAG));
+    }
+
+    /**
+     * Returns a {@link BasicTypes#BASIC_METACARD} version of the given
+     * {@link HistoryMetacardImpl}
+     *
+     * @return The converted metacard
+     * @throws IllegalStateException
+     */
+    public Metacard toBasicMetacard() {
+        String id = this.getIdHistory();
+        if (id == null || id.matches("\\s*")) {
+            throw new IllegalStateException(
+                    "Cannot convert history metacard without the original metacard id");
+        }
+
+        MetacardImpl result = new MetacardImpl(this, BasicTypes.BASIC_METACARD);
+        result.setId(id);
+        result.setTags(this.getTagsHistory());
+        return result;
+    }
+
+    public String getIdHistory() {
+        return requestString(ID_HISTORY);
+    }
+
+    public void setIdHistory(String idHistory) {
+        setAttribute(ID_HISTORY, idHistory);
+    }
+
+    public Date getVersioned() {
+        return requestDate(VERSIONED);
+    }
+
+    public void setVersioned(Date date) {
+        setAttribute(VERSIONED, date);
+    }
+
+    public String getEditedBy() {
+        return requestString(EDITED_BY);
+    }
+
+    public void setEditedBy(String editedBy) {
+        setAttribute(EDITED_BY, editedBy);
+    }
+
+    public Action getState() {
+        return Action.fromKey(requestString(STATE));
+    }
+
+    public void setState(Action state) {
+        setAttribute(STATE, state.getKey());
+    }
+
+    public Set<String> getTagsHistory() {
+        Attribute attribute = getAttribute(TAGS_HISTORY);
+        if (attribute == null || attribute.getValue() == null) {
+            return new HashSet<>();
+        } else {
+            return new HashSet<>(getAttribute(TAGS_HISTORY).getValues()
```

Comment:
You're calling `getAttribute(TAGS_HISTORY)` again when you already have `attribute`.